### ISSUE TITLE
v702: 'Unexpected word "EXPORTING" in functional method call' syntax error not detected

### ISSUE THREAD

**Issue by fdupret** [abaplint/abaplint#3634]
This snippet yields syntax errors in a 702 system that are not detected by abaplint (neither `parser_702_chaining` or `check_syntax` or `parser_error`).
```abap
REPORT zfoo.

CLASS lcl DEFINITION.
  PUBLIC SECTION.
    CLASS-METHODS method
      IMPORTING
        iv_value        TYPE i
      RETURNING
        VALUE(rv_value) TYPE i.
ENDCLASS.

DATA foo TYPE i.

foo = lcl=>method( 42 ). " OK
foo = lcl=>method( iv_value = 42 ). " OK
foo = lcl=>method( EXPORTING iv_value = 42 ). " KO: Unexpected word "EXPORTING" in functional method call
lcl=>method( EXPORTING iv_value = lcl=>method( 42 ) ). " OK (surprisingly)
lcl=>method( EXPORTING iv_value = lcl=>method( EXPORTING iv_value = 42 ) ). " KO: Unexpected word "EXPORTING" in functional method call


CLASS lcl IMPLEMENTATION.
  METHOD method.
  ENDMETHOD.
ENDCLASS.
```

Generally speaking, in a v702 system, it is not allowed to mix procedural calls with functional calls in an assignment (not necessarily the whole statement).

As a workaround, we use `exporting` rule to detect such cases even though it yields more results than needed (statement number 4 here is syntactically correct).

**Comment by larshp** [comment]
for `foo = lcl=>method( EXPORTING iv_value = 42 ).` it detects that exporting can be omitted, but you expect something else?

<img width="814" height="496" alt="Image" src="https://github.com/user-attachments/assets/43b7a89d-442e-48ef-8efd-88350ca426f5" />

**Comment by fdupret** [comment]
No that's fine and so far `exporting` has worked exactly as we would expect. The issue is more in using it as a workaround and that's on us.
To better explain our use of abaplint: for now we're only using it to detect downport issues. We're not interested (yet) in the stylistic part. In our case we had 23 findings with `exporting` but only one of them needed to be addressed to solve 702 compatibility. For example we had statements such as `lcl=>method( EXPORTING iv_value = 42 ).` highlighted by `exporting` but they're syntactically correct. They don't _require_ a fix, it's "only" a stylistic issue.

TL;DR: this issue is about `parser_error` or `check_syntax` in 702, nothing wrong with `exporting`.

**Comment by larshp** [comment]
extend parser_702_chaining
